### PR TITLE
Disable signing except for master and maintenance branches

### PR DESCRIPTION
**Pull request by merks**
https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/794

### CODE REVIEW

**Comment by merks** [comment]
@akurtakov 

It's like this failure came back:

```
16:06:05  [ERROR] Cannot resolve project dependencies:
16:06:05  [ERROR]   Software being installed: eclipse-junit-tests 4.27.0.qualifier
16:06:05  [ERROR]   Missing requirement: org.eclipse.test.feature.group 3.8.400.v20230109-1044 requires 'org.eclipse.equinox.p2.iu; org.mockito.mockito-core [4.10.0,4.10.0]' but it could not be found
16:06:05  [ERROR]   Cannot satisfy dependency: eclipse-junit-tests 4.27.0.qualifier depends on: org.eclipse.equinox.p2.iu; org.eclipse.test.feature.group 0.0.0
```
Even though I bumped the version:

https://github.com/eclipse-platform/eclipse.platform.releng/pull/176

And the build for this passed afterwards:

https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/pull/791

Is it possible that a new IBuild "used the qualifier up" before the TP changes were committed and now the version needs to change again?

**Comment by akurtakov** [comment]
> Is it possible that a new IBuild "used the qualifier up" before the TP changes were committed and now the version needs to change again?

Absolutely, you have to merge the target platform change and the touch in the same I-build otherwise it would not work.

**Comment by merks** [comment]
Wow, what a nightmare.  One never knows when an IBuild might kick in between the two.  Oh well, such is life...

**Comment by merks** [comment]
@akurtakov 

The build time is reduced to 17 minutes:

image

versus a more typical 45 minutes:

image

If I screwed something up, which I doubt, it's possible that the master build won't do signing, so we need to keep an eye on that.

The most recent master build failed

https://ci.eclipse.org/platform/job/eclipse.platform.releng.aggregator/job/master/887/

for the same reason that this PR's build failed earlier:

https://ci.eclipse.org/platform/job/eclipse.platform.releng.aggregator/job/PR-795/2/

Do you have any concerns about merging this change?

**Comment by akurtakov** [comment]
Considering that this master build is not used for anything (in comparison to m2e one which pushes to snapshots) I don't see any reason to not merge it now.

File: Jenkinsfile [hunk 1]
```diff
@@ -42,8 +42,16 @@ pipeline {
 			steps {
 				withCredentials([string(credentialsId: 'gpg-passphrase', variable: 'KEYRING_PASSPHRASE')]) {
 					sh '''
+					if [[ ${BRANCH_NAME} == master ]] || [[ ${BRANCH_NAME} =~ ^R[0-9]_[0-9]+_maintenance ]]; then
+						MVN_ARGS="-Peclipse-sign"
+					else
+						MVN_ARGS=
+						export KEYRING="deadbeef"
```
Comment (akurtakov) [inline Jenkinsfile:49]: This looks weird. Why is it needed at all?

Comment (merks) [reply Jenkinsfile:49]: Which of these 4 lines specifically are of concern? 

The ${MVN_ARGS} is used later so either the option `-Peclipse-sign` is present or it's not present in the mvn command line.  The KEYRING environment variable contains the **private key** and it's better, I think, that the private key not be in the environment where it is visible in any running Java tests, for example, such that someone might have access to it (save it to a file that's archived) and later try to crack the passphrase to make use of it.  Alternatively we could guard the `'initialize PGP'` stage with the same guard as the `'Deploy eclipse-platform-parent pom and eclipse-sdk target'` stage such that it's not placed in the environment in the first place:
```
			when {
				anyOf {
					branch 'master'
					branch 'R*_maintenance'
				}
			}
```
But that's more code duplication.  Or we could "just not care" and remove line 49.  But as I said, I'm not sure specifically which of these lines look "weird"...

Comment (akurtakov) [reply Jenkinsfile:49]: I am speaking for the export KEYRING="deadbeef" part.

**Comment by akurtakov** [comment]
Let's merge and be done with it.

**Pull request merged**